src/kernel/arch/x86/paging.zig

```diff
@@ -94,15 +149,91 @@ fn mapDir(dir: *Directory, phys_start: u32, phys_end: u32, virt_start: u32, virt
         return PagingError.UnalignedVirtAddresses;
     }
 
+    const entry = virt_start / PAGE_SIZE_4MB;
+    var dir_entry = &dir.entries[entry];
+    dir_entry.* |= DENTRY_PRESENT;
+    dir_entry.* |= DENTRY_WRITABLE;
+    dir_entry.* &= ~u32(DENTRY_USER);
+    dir_entry.* |= DENTRY_WRITE_THROUGH;
+    dir_entry.* &= ~u32(DENTRY_CACHE_DISABLED);
+    dir_entry.* &= ~u32(DENTRY_4MB_PAGES);
+
+    // Only create a new table if one hasn't already been created for this dir entry.
+    // Prevents us from overriding previous mappings.
+    var table: *Table = undefined;
+    if (dir.tables[entry]) |tbl| {
```

Comment:
Same here, check out of bounds. Unless it is impossible, then this is fine.

Reply:
Yeah this is covered by the previous index check :)